emcc.py

```diff
@@ -1357,6 +1357,15 @@ def check(input_file):
       if not shared.Settings.WASM:
         options.memory_init_file = True
 
+    if shared.Settings.WASM and not shared.Settings.WASM_BACKEND:
+      # NB: asm2wasm auto-generates code that imports setTempRet0/getTempRet0
+      #     so make sure to include those.
+      #     this applies even if MINIMAL_RUNTIME is set
+      shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE.extend([
+        'setTempRet0',
+        'getTempRet0',
```

Comment:
I don't understand why this is needed as part of this change.

Reply:
It's a bit involved.

asm2wasm generates imports of setTempRet0 and getTempRet0 when generating code. Previously these imports were bound to the `support.js` versions of those functions (imported via `Settings.RUNTIME_FUNCS_TO_IMPORT`), not the `library.js` versions. Since the `library.js` versions no longer use a "_" prefix and use the import name that the `support.js` previously used, they must now be forced to import included.

Reply:
@kripken can you take a look at this?  I'm not sure I understand the original intended behaviour here.

Reply:
This part should be ok. But then we should remove the support.js ones entirely.

Reply:
In addition to fixing support.js, the comment here can be `asm2wasm may auto-generates` - adding "may", since it depends on what optimizations are run.